### PR TITLE
 [llama] Llama2/3 tests added to CI

### PR DESCRIPTION
**Pull request by kevin-mii**
- [x]  llama2/3 experimental model + unit tests added to CI, fixed issues with attention and decoder prepare_inputs
- [x] Add fuse_batch option to matmul_2d to fit longer seq_len matmuls

### CODE REVIEW

**Review by tapspatel — APPROVED** [review]

**Comment by tapspatel** [comment]
Approved pending successful pipeline run

**Comment by mikevin920** [comment]
T3k Frequent CI: https://github.com/tenstorrent/tt-metal/actions/runs/9422180348
Post-commit: https://github.com/tenstorrent/tt-metal/actions/runs/9422184713

**Review by tt-rkim — APPROVED** [review]

**Comment by mikevin920** [comment]
T3K Pipeline: https://github.com/tenstorrent/tt-metal/actions/runs/9471525297
Post-commit: https://github.com/tenstorrent/tt-metal/actions/runs/9471531645

**Review by ttmchiou — APPROVED** [review]

**Review by TT-BrianLiu — APPROVED** [review]

File: models/experimental/llama2_70b/tests/test_llama_mlp_t3000.py [hunk 1]
```diff
@@ -8,25 +8,68 @@
 from models.utility_functions import get_devices_for_t3000, skip_for_grayskull
 from models.experimental.llama2_70b.tests.test_llama_mlp import run_test_LlamaMLP_inference
 
+import os
+
+# Set Llama flags for CI, if CI environment is setup
+if os.getenv("CI") == "true":
```
Comment (tt-rkim) [inline models/experimental/llama2_70b/tests/test_llama_mlp_t3000.py:14]: Do you mind taking this opportunity to move this `os.getenv("CI") == "true"` logic that's used everywhere and put it into a utility function for tests?

Comment (mikevin920) [reply models/experimental/llama2_70b/tests/test_llama_mlp_t3000.py:14]: Will move them into a common function in next PR

**Pull request merged**